reinsurancecontract.py

```diff
@@ -63,7 +63,19 @@ def mature(self, time):
         else: #TODO: ? Instead: if self.insurancetype == "proportional":
             self.contract.unreinsure()
 
-        if np.random.uniform(0,1,1) < 0.95:
+        if np.random.uniform(0,1,1) < 1:
             reinrisk = self.property_holder.create_reinrisk(time,self.category)
-            if reinrisk is not None and hasattr(self.insurer, 'reinrisks_kept'):
-                self.insurer.reinrisks_kept.append(reinrisk)
\ No newline at end of file
+
+            if reinrisk is not None:
+                new_nonproportional_risks = [reinrisk]
+                [reinrisks_per_categ, number_reinrisks_categ] = self.insurer.risks_reinrisks_organizer(new_nonproportional_risks)  #Here the new reinrisks are organized by category.
```

Comment:
Please do not use inline comments unless absolutely necessary. Instead use comments contained in three quotation marks in the line before the respective line of code to describe normal program logic.

Reply:
OK!